tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,338 @@
+    percentage: the percentage of layers to freeze
+  """
+  if percentage < 0 or percentage > 1:
+    raise Exception("freeze percentage should between 0.0 and 1.0")
+
+  num_layers = len(model.layers)
+  num_layers_to_freeze = int(num_layers * percentage)
+  for idx, layer in enumerate(model.layers):
+    if idx < num_layers_to_freeze:
+      layer.trainable = False
+    else:
+      layer.trainable = True
+
+def _build_keras_model() -> tf.keras.Model:
+  """Creates a Image classification model with MobileNet as backbone
+
+  Returns:
+    The image classifcation Keras Model and the backbone MobileNet model
+  """
+  # We create a MobileNet model with weights pre-trained on ImageNet.
+  # We remove the top classification layer of the MobileNet, which was
+  # used for classifying ImageNet objects. We will add our own classification
+  # layer for CIFAR10 later. We use average pooling at the last convolution
+  # layer to get a 1D vector for classifcation, which is consistent with the
+  # origin MobileNet setup
+  base_model = tf.keras.applications.MobileNet(
+      input_shape=(224, 224, 3), include_top=False, weights='imagenet',
+      pooling='avg')
+
+  # We add a Dropout layer at the top of MobileNet backbone we just created to prevent
+  # overfiting, and then a Dense layer to classifying CIFAR10 objects
+  model = tf.keras.Sequential([
+      tf.keras.layers.InputLayer(
+          input_shape=(224, 224, 3), name=transformed_name(IMAGE_KEY)),
+      base_model,
+      tf.keras.layers.Dropout(0.3),
+      tf.keras.layers.Dense(10, activation='softmax')
+  ])
+
+  return model, base_model
+
+def _data_augmentation(feature_dict):
+  """Perform data augmentation on batches of data.
+
+  Args:
+    feature_dict: a dict containing features of samples
+
+  Returns:
+    The feature dict with augmented features
+  """
+  image_feature = feature_dict[transformed_name(IMAGE_KEY)]
+  batch_size = tf.shape(image_feature)[0]
+  image_feature = tf.image.random_flip_left_right(image_feature)
+  image_feature = tf.image.resize_with_crop_or_pad(image_feature, 250, 250)
+  image_feature = tf.image.random_crop(image_feature, (batch_size, 224, 224, 3))
+  feature_dict[transformed_name(IMAGE_KEY)] = image_feature
+  return feature_dict
+
+# TFX Transform will call this function.
+def preprocessing_fn(inputs):
+  """tf.transform's callback function for preprocessing inputs.
+
+  Args:
+    inputs: map from feature keys to raw not-yet-transformed features.
+
+  Returns:
+    Map from string feature key to transformed feature operations.
+  """
+  outputs = {}
+
+  # tf.io.decode_png function cannot be applied on a batch of data
+  image_features = tf.map_fn(lambda x: tf.io.decode_png(x[0], channels=3),
+                             inputs[IMAGE_KEY], dtype=tf.uint8)
+  image_features = tf.cast(image_features, tf.float32)
+  image_features = tf.image.resize(image_features, [224, 224])
+  image_features = tf.keras.applications.mobilenet. \
```

Comment:
lets make this one line and break at image_features. that is:
tf.keras.applications.mobilenet.preprocess_input(
   image_features)